scripts/products/generate-setProducts-tx.js

```diff
@@ -0,0 +1,95 @@
+require('dotenv').config();
+const { ethers, config } = require('hardhat');
+const fs = require('fs');
+const path = require('path');
+
+const { parse: csvParse } = require('csv-parse/sync');
+
+const { MaxUint256 } = ethers.constants;
+
+const OUTPUT_FILE = path.join(
+  config.paths.root,
+  'scripts/products', // dir
+  'setProducts-txs.json', // filename
+);
+
+const YIELD_TOKEN_PRODUCT_TYPE_ID = '2';
+
+/**
+ *
+ * Generate the tx data for the Cover.setProducts transaction based using the data
+ * in the productsDataFile CSV.
+ *
+ * CLI parameters in this order: productsDataFilePath coverAddress signerAddress
+ *
+ * NOTE: Product editing not yet supported. 'Product Id' is ignored.
+ *
+ * @param provider
+ * @param productsDataFilePath path for file of products
+ * @param coverAddress address of the Cover contract
+ * @param signerAddress address of the AB signer - not encoded in the transaction blob
+ * @returns {Promise<{setProductsTransaction: *}>}
+ */
+const main = async (provider, productsDataFilePath, coverAddress) => {
+  console.log(`Using cover address: ${coverAddress}.`);
+
+  const { abi } = JSON.parse(fs.readFileSync('./artifacts/contracts/modules/cover/Cover.sol/Cover.json'));
+
+  const cover = new ethers.Contract(coverAddress, abi);
```

Comment:
can't you just use artifacts like this:
`const cover = artifacts.require('Cover');`, where `artifacts` is improted from `hardhat`?

Reply:
we can't get rid of bit but now that i think about it let's hardcode it - it's a proxy contract, this never changes unless we're rearchitecting things substantially.